provider/provider.go

```diff
@@ -164,24 +164,26 @@ func (clientTLS *ClientTLS) CreateTLSConfig() (*tls.Config, error) {
 	cert := tls.Certificate{}
 	_, errKeyIsFile := os.Stat(clientTLS.Key)
 
-	if _, errCertIsFile := os.Stat(clientTLS.Cert); errCertIsFile == nil {
-		if errKeyIsFile == nil {
-			cert, err = tls.LoadX509KeyPair(clientTLS.Cert, clientTLS.Key)
-			if err != nil {
-				return nil, fmt.Errorf("Failed to load TLS keypair: %v", err)
+	if len(clientTLS.Cert) > 0 && len(clientTLS.Key) > 0 {
```

Comment:
This confition will allow user to specify TLS configuration with no key r cert file even if the `InsecureSkipVerify` parameter is set to `false`.

It can be better to check the `InsecureSkipVerify` in the condition and let the current mechanism generate an error if cert file or key file are missing when `InsecureSkipVerify` is set to false.

```go
if ! clientTLS.InsecureSkipVerify {
...
}
```